src/bp-activity/bp-activity-functions.php

```diff
@@ -3508,6 +3508,16 @@ function bp_activity_delete_children( $activity_id, $comment_id ) {
 	if ( ! empty( $children ) ) {
 		foreach ( (array) $children as $child ) {
 			bp_activity_delete_children( $activity_id, $child->id );
+
+			/**
+			 * Fires after deleting all the chidren comment of comment.
+			 *
+			 * @since BuddyPress 2.0.0
+			 *
+			 * @param int   $activity_id Activity item id.
+			 * @param array $children    array of children item of comment.
+			 */
+			do_action( 'bp_activity_delete_comment_children', $activity_id, $child->id );
```

Comment:
@janakkaneriya007 Above action is not needed as `bp_activity_delete_children` function use `bp_activity_delete` function. That funcation has hooks & filter.

Reply:
@dipskakadiya We need to keep this as it iterates with all the child comments of the activity and any of those chid comments is moderated then we must need to delete it and it's the last hook where we are getting children comments